### PR TITLE
Improve HOC map design

### PR DESCRIPTION
**Pull request by bethanyaconnor**
Update HOC map to use design Mark created. It will need to use a style created by us and hosted on Mapbox to have access to the background and icons.

Screenshot 2019-09-04 at 3 23 23 PM

This PR includes a couple of other small changes to get it ready for launch.

##OLD Description
Mapbox allows you to style a map, including the data, on their site and just use that "style". I've been using that feature to prototype designs and think we should use this in the HOC map in production (tbd on the census map).

Reasons for this change:
 1. Because we're going to start using custom icons, we'd need to import a style anyway (easiest way to expose these custom icons)
 2. We don't need to wait for a deploy for style changes
 3. It allows us to [download an image of the map](ttps://docs.mapbox.com/api/maps/#static-images) to be served statically during hour of code

Downsides to this change:
 1. The legend is still defined using HTML/CSS and therefore will be _impossible_ to keep in sync if we change color schemes.
 2. Anyone with access to the style can change the look of the map in production without review.

The biggest driver for this change is reason number 3 for me. Notably, I don't know of any way of displaying a static image instead of the map during Hour of Code without this change, other than duplicating the JS defined style in a Mapbox style and using that.

This PR also contains a change to how popups are shown for special events to be more consistent with the current map.  I can separate that out if that's problematic.

### CODE REVIEW

File: apps/src/sites/hourofcode.com/pages/views/hoc_events_map_replacement.js [hunk 1]
```diff
@@ -3,61 +3,70 @@
...
-        ],
-        'circle-stroke-width': 1,
-        'circle-stroke-color': '#000000'
+    map.on('click', 'hoc-events', function(e) {
+      var coordinates = e.features[0].geometry.coordinates.slice();
+      var organization_name = e.features[0].properties.organization_name;
+
+      // Ensure that if the map is zoomed out such that multiple
+      // copies of the feature are visible, the popup appears
+      // over the copy being pointed to.
```
Comment (breville) [inline apps/src/sites/hourofcode.com/pages/views/hoc_events_map_replacement.js]: Not quite sure what this comment means...

File: apps/src/sites/hourofcode.com/pages/views/hoc_events_map_replacement.js [hunk 2]
```diff
@@ -3,61 +3,70 @@
...
+      while (Math.abs(e.lngLat.lng - coordinates[0]) > 180) {
```
Comment (breville) [inline apps/src/sites/hourofcode.com/pages/views/hoc_events_map_replacement.js]: Is it worth de-duping this with the similar logic just below?

File: apps/src/sites/hourofcode.com/pages/views/hoc_events_map_replacement.js [hunk 3]
```diff
@@ -3,61 +3,70 @@
 $(document).ready(function() {
   var map = new mapboxgl.Map({
     container: 'mapbox-map',
-    style: 'mapbox://styles/mapbox/streets-v11',
+    style: 'mapbox://styles/codeorg/cjz36duae88ds1cp7ll7smx6s',
```
Comment (breville) [inline apps/src/sites/hourofcode.com/pages/views/hoc_events_map_replacement.js]: Should we hoist this magic number into a variable with an explanation?

Comment (bethanyaconnor) [reply apps/src/sites/hourofcode.com/pages/views/hoc_events_map_replacement.js]: I pulled this out and added a brief comment but not sure what else to say.

File: apps/src/sites/hourofcode.com/pages/views/hoc_events_map_replacement.js (hunk 1 above)
Comment (bethanyaconnor) [reply apps/src/sites/hourofcode.com/pages/views/hoc_events_map_replacement.js]: The logic is correcting for the potential of a feature/point being on the map twice (when it's zoomed out enough) and ensuring that the popup appears where the click happened. I took this logic from Mapbox.

It's very easy to simplify to make it just show the popup on the spot where the user clicked but I think it looks a bit nicer for the popup to be placed on the point itself.

File: apps/src/sites/hourofcode.com/pages/views/hoc_events_map_replacement.js (hunk 2 above)
Comment (bethanyaconnor) [reply apps/src/sites/hourofcode.com/pages/views/hoc_events_map_replacement.js]: Deduplicated a bit of the logic. There are a couple of subtle differences between the two listeners but let me know if there's anything else you'd like to see pulled out!

File: apps/src/sites/hourofcode.com/pages/views/hoc_events_map_replacement.js [hunk 4]
```diff
@@ -1,9 +1,11 @@
 /*global mapboxgl*/
 
 $(document).ready(function() {
+  // We keep some style elements as a Mapbox style for simplicity.
+  const style_path = 'mapbox://styles/codeorg/cjz36duae88ds1cp7ll7smx6s';
```
Comment (breville) [inline apps/src/sites/hourofcode.com/pages/views/hoc_events_map_replacement.js]: nit: `stylePath`

File: apps/src/sites/hourofcode.com/pages/views/hoc_events_map_replacement.js [hunk 5]
```diff
@@ -13,43 +15,52 @@ $(document).ready(function() {
 
   map.dragRotate.disable();
 
+  // Ensure that if the map is zoomed out such that multiple
+  // copies of the point are visible, the popup appears
+  // over the point clicked on.
+  function getPopupCoordinates(clickLngLat, featureCoordinates) {
+    var normalizedCoordinates = featureCoordinates;
```
Comment (breville) [inline apps/src/sites/hourofcode.com/pages/views/hoc_events_map_replacement.js:22]: nit: `'let`

File: apps/src/sites/hourofcode.com/pages/views/hoc_events_map_replacement.js [hunk 6]
```diff
@@ -13,43 +15,52 @@ $(document).ready(function() {
...
-        coordinates[0] += e.lngLat.lng > coordinates[0] ? 360 : -360;
-      }
       if (popup) {
         popup.remove();
       }
-      popup = new mapboxgl.Popup()
-        .setLngLat(coordinates)
-        .setText(organization_name)
+      const popupText =
+        organization_name + (city.length > 0 ? ' (' + city + ')' : '');
```
Comment (breville) [inline apps/src/sites/hourofcode.com/pages/views/hoc_events_map_replacement.js]: nit: `organizationName`

File: apps/src/sites/hourofcode.com/pages/views/hoc_events_map_replacement.js (hunk 5 above)
Comment (breville) [reply apps/src/sites/hourofcode.com/pages/views/hoc_events_map_replacement.js:22]: oh, are we not going through the preprocessor for this JS file?  In that case, disregard.

File: apps/src/sites/hourofcode.com/pages/views/hoc_events_map_replacement.js (hunk 1 above)
Comment (breville) [reply apps/src/sites/hourofcode.com/pages/views/hoc_events_map_replacement.js]: Ah, sounds good.

**Comment by bethanyaconnor** [comment]
Loading the data in JS instead of as a Mapbox style is in 5c2193282c475dbf4388ed537ec16b7af018a526. In this case we would have to screenshot the map a few days before Hour of Code (which TIL is what we've generally done). Thoughts?

**Comment by breville** [comment]
It sounds like taking the screenshot manually is the least burdensome approach.

**Comment by bethanyaconnor** [comment]
Sounds good! I've updated the description and title to reflect what this PR now does.

**Review by breville — APPROVED** [review]

**Pull request merged**